examples/rag/state_of_the_union_rag.py

```diff
@@ -0,0 +1,51 @@
+import os
+
+from dotenv import load_dotenv
+
+from mirascope.base import BaseChunker
+from mirascope.base.types import BaseChunkerParams
+from mirascope.chroma.vectorstores import ChromaVectorStore
+from mirascope.openai import OpenAICall, OpenAIEmbedder
+
+load_dotenv()
+
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+class Chunker(BaseChunker):
+    chunker_params = BaseChunkerParams(chunk_size=1000, chunk_overlap=200)
+
+
+class SOTU(ChromaVectorStore):
+    embedder = OpenAIEmbedder()
+    chunker = Chunker()
+    index_name = "sotu"
+
+
+class SotuCall(OpenAICall):
+    prompt_template = """
+    SYSTEM: Answer the question based on the context.
+    {context}
+    USER: {question}
+    """
+
+    question: str
+
+    @property
+    def context(self):
+        with open(f"{current_dir}/state_of_the_union.txt") as file:
```

Comment:
this should already be done separately outside of the property, right?

Instead user should define a vector store field and use that, right? e.g.

```python
@property
def context(self) -> list[str]:
    return [doc.text for doc in self.store.query(self.query)]
```

Reply:
this is because this is a non-persistent chroma instance. so if you use `PersistentClient` or `HttpClient` then you are correct

Reply:
But even if it isn't persistent, it's still persistent in memory right? The user won't want to load each document every time the call is made. And using a computed field / cached_property here doesn't save us since it will make a new call if anything (e.g. a field) has changed.

Can a user not define a non-persistent index and pass that in as an attribute? That way they only have to create the index once?

Reply:
this is scratch work, the final example will not use the normal chroma client